compiler/generator_ts.c

```diff
@@ -0,0 +1,1308 @@
+
```

Comment:
This file is very similar to `generator_js.c` - from a maintenance point of view I think it would make more sense to share a single generator implementation between javascript and typescript and just parameterise the small number of differences - that way future fixes only need applying in one place rather than two (if there are two places to fix, inevitably one gets missed sometimes).

Also, it seems many of the differences are due to trying to format the generated code to match typescript code formatting conventions.  I'm not sure how worthwhile a goal that is - having generated code with a basically readable formatting to aid debugging generator problems is useful, but coding conventions almost universally aim at making code more maintainable, and the generated code inherently doesn't need to be maintained directly.

So I'd be inclined to just tell typescript to shut up about code formatting (with more/broader `eslint-disable` magic comments), but if that's not possible we could format the javascript output in the same way.

There's also the approach the Go generator takes which is to run the snowball `.go` outputs through `gofmt` to mechanically reformat them to match Go coding conventions.  That has the advantage that there's no snowball-specific work to do if new formatting checks get added.